test/txsim/client.go

```diff
@@ -227,6 +227,9 @@ func (tc *TxClient) Broadcast(ctx context.Context, txBuilder sdkclient.TxBuilder
 	}
 
 	for {
+		if ctx.Err() != nil {
+			return nil, ctx.Err()
+		}
 		subctx, cancel := context.WithTimeout(ctx, rpcContextTimeout)
 		defer cancel()
```

Comment:
also, this is the change that seems to fix the issue in #1782 since its no longer occuring and this seems to make sense, but would defer to @cmwaters on if we should be doing this

Reply:
is there error check suppose to also catch the context `Done()` event?

if so, shouldn't we just do this:
```go
for {
  select {
    case: <- ctx.Done()
        return nil, ctx.Err()
    default:
  }
  ...
}
```

little bit of a nit, but I think this is more clear that we are managing the for loop based on the ctx being active.